### PR TITLE
Skip workflows without external ID [SSW-650]

### PR DESCRIPTION
**Pull request by erosdome**
### Checklist

- [ ] I've read and followed the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- [ ] `step.yml` and `README.md` is updated with the changes (if needed)

### Version

Requires a *MAJOR/MINOR/PATCH* [version update](https://semver.org/)

### CODE REVIEW

File: step/build_id_getter.go [hunk 1]
```diff
@@ -68,6 +68,9 @@ func (bg BuildIDGetter) createKeyValuePairSlice() []keyValuePair {
 	var stageWorkflowMap []keyValuePair
 	for _, stage := range bg.FinishedStages {
 		for _, wf := range stage.Workflows {
+			if wf.ExternalId == "" {
```
Comment (morbalint) [inline step/build_id_getter.go:71]: shouldn't we log this?

Comment (zsolt-vicze) [reply step/build_id_getter.go:71]: Plus one on this one. I strongly prefer to log something to the customer so they can see in the logs what's going on.

File: step/build_id_getter.go [hunk 2]
```diff
@@ -68,6 +71,10 @@ func (bg BuildIDGetter) createKeyValuePairSlice() []keyValuePair {
...
+				bg.logger.Printf("Empty external ID for workflow %s in stage %s", wf.Name, stage.Name)
```
Comment (zsolt-vicze) [inline step/build_id_getter.go:75]: I don't think customers will/should know anything about external ID. Can we log something like this:
`Skipping workflow %s in stage %s. Workflow was not executed.`?
Also, I would wait a confirmation from a mobile engineer to make sure this is the right way to log information for the customer.

Comment (morbalint) [reply step/build_id_getter.go]: @godrei can you please confirm the logging function?

**Review by zsolt-vicze — COMMENTED** [review]
It looks good to me but I think we really should have a mobile-expert eye on this.
E.g. I am not sure
- if that function is the best place to filter out these. Maybe the caller should do the filtering?
- if this is the right way to log information for the customer.

**Comment by godrei** [comment]
@zsolt-vicze regarding 
> if that function is the best place to filter out these. Maybe the caller should do the filtering?

I think yes with the current architecture of the step. BuildIDGetter encapsulates the functionality of listing build ids for later functionalities, so ensuring we list valid build ids is fine here.

**Review by godrei — APPROVED** [review]

**Pull request merged**